### PR TITLE
Remove the (unmaintained) Autotools build method.

### LINKED ISSUE #4461: Drop Autotools, move to CMake
As you may have seen from my struggles in #4459, autotools is very finicky about certain things, difficult to maintain, and prone to occasional breakage. Having two build systems is decidedly sub-optimal in that it slows down anyone writing new code, as they have to know both build systems _and_ take the time to update and resolve any bugs arising in either system.

Given the conundrum, I propose that we drop autotools in favor of CMake. The reasoning for this is threefold:
- Autotools is old (not necessarily a con), complicated to configure, requires manually listing the sources and headers, very finicky about whitespace, and requires quite a bit of dev time to maintain. It's also *extremely* verbose, at 1721 total lines of configuration - counting *only* Makefile.am and configure.ac.
- CMake has much better C/C++ support, including automatic header discovery, better linking and CFLAGS support, and is much less verbose and easier to configure, coming in at only 262 lines.
- CMake *also* has much better build tool support, generating Ninja, Make, and even Visual Studio build configurations. It also has built-in support for generating `compile_commands.json`, allowing code completion in editors without built-in C/C++ support, like vim or Atom.

From my perspective, I'm not seeing any reason to keep autotools around. If any of you have objections or new information, please voice them soon, as I'm going to start work on implementing this by next week at latest.

As an aside, @fluffyfreak you do know you can use CMake to generate the VS solutions, right? I'm in no way a Windows dev, but it seems we might be able to remove the need to manually update the windows build files by automating it.

### LINKED ISSUE #4504: configure --with-strip fails
### Observed behaviour

When configuring pioneer with `./configure --with-strip` it will fail.
```
config.status: executing depfiles commands
config.status: error: in `/tmp/pioneer':
config.status: error: Something went wrong bootstrapping makefile fragments
    for automatic dependency tracking.  Try re-running configure with the
    '--disable-dependency-tracking' option to at least be able to build
    the package (albeit without support for automatic dependency tracking).
See `config.log' for more details
```
Full log: https://pastebin.com/x9ERwCPP

If `--with-strip` is omitted or if `--disable-dependency-tracking` is also used then configure will succeed, but neither is ideal. 

I bisected this and found the first bad commit.
```
ec70cd27bd7e3d50db97c07f85ea8693b22dad25 is the first bad commit
commit ec70cd27bd7e3d50db97c07f85ea8693b22dad25
Author: Webster Sheets <webster@web-eworks.com>
Date:   Sat Sep 29 12:15:56 2018 -0400

    Fix some crashes relating to loading old saves.
    
    Catch JSON type errors in LuaGame.cpp.
    
    Apparently, Autotools doesn't like tabs. Who'da thunk it?
    
    Fixup the saved game loading code.
    
    Fix save loading errors, patch JSON library to throw instead of asserting.
    
    Theoretically finally fix the serialization code.
    
    String lifespan fix in unpickle_json

:040000 040000 cd403501c8b38853d57d934faace3117390a7b88 49c367d279268936e84b145c73078f578b04f0d4 M	contrib
:040000 040000 32b9b72d164525ed7d0207b2357bfbc485cc57b4 d382017f500705624cf2fbfde433c0a6dcd32f5d M	data
:040000 040000 4e44c821d5228d390c1c36b7aac00c8b0ee92bcc 625dccd4d10702c1dda2a9418cfa1a0fad49998b M	src
```
ec70cd27bd7e3d50db97c07f85ea8693b22dad25

### Expected behaviour

pioneer should configure with `--with-strip` enabled.

### Steps to reproduce

1. `./bootstrap`
2. `./configure --with-strip`

__My pioneer version (and OS):__

pioneer: https://github.com/pioneerspacesim/pioneer/commit/3fdd945ee43ae73b43cf61f8f41e6cf6eeb31f9a
OS: `Slackware64-current`

### PR DESCRIPTION
**Pull request by Web-eWorks**
Closes #4461, closes #4504.

This PR finishes off the series of CMake improvements. CMake is now stable enough that it's the main compilation method for non-Windows platforms, and Travis is now exclusively using CMake as it's build tool. Considering the issues we have with maintaining the autotools port (no one wants to do it, it's 1700 lines of DSL, etc...) I feel it is time that we finally struck support for Autotools from the project entirely.

You are welcome to consider this a happy New Year's gift. :smile:

### CODE REVIEW

**Comment by Web-eWorks** [comment]
You're right - I grabbed the wrong issue number.

**Comment by fluffyfreak** [comment]
@Philbywhizz how will this affect the OSX build? Are you able to use CMake to do them?

**Comment by Philbywhizz** [comment]
Sorry, I'm not really familiar with CMake.

**Comment by impaktor** [comment]
I think this needs some decision now/soonish. Basically, on the Linux side, this is good to go. 

I don't know how/if autotools is needed for OSX or Windows builds, and what the status is for Travis/appveryor builds, do they need autotools?

ping: @pcercuei (you were the appveyor knowledgeable person right?) @fluffyfreak  

...and once that's done, conflicts in this branch needs fixing,either by author or merger.

**Comment by fluffyfreak** [comment]
I know that there was an effort to get OSX builds working via Travis, but I don't know how far along it is. 

That's really the only remaining hold up to removing this stuff... though it should probably go now anyway since it's been unmaintained for so long.

**Comment by pcercuei** [comment]
Windows builds are done with CMake.

**Comment by Web-eWorks** [comment]
@fluffyfreak Check literally any travis build from the past month and a half - it's done and dusted. OSX packaging is another thing, but that's simply a matter of adjusting some locations in the package script manually run by @Philbywhizz.

Travis is good to go; it's all been moved to CMake and we're doing OSX CI builds with CMake as well. Manual windows builds are done with Fluffy's VS project files, which don't use autotools. Appveyor with CMake looks good too, so there's really no reason to keep autotools around any more.

When I get a signoff I'll bring this PR back in line with master so we can merge it - hopefully it won't take until November to get this moving.

**Comment by impaktor** [comment]
I/we'll merge as soon as conflict is resolved. If it goes beyond november, I'll resolve the conflict. I think it's just something trivial in COMPILING.txt, someone added MSYS2 instructions.

**Comment by Web-eWorks** [comment]
@impaktor ready for merge! Best, most belated new-year's gift I've ever given.

File: COMPILING.txt [hunk 1]
```diff
@@ -28,13 +23,11 @@ Table of Contents
...
-       CMake with GCC or Clang
+Linux: CMake with GCC or Clang
 Windows: Microsoft Visual C++ 2015 (Community or Pro)
          Microsoft Visual C++ 2017 (Community or Pro)
          CMake with MSYS2 (and MinGW-w64)
-Windows: GNU Autotools with MXE (MinGW GCC) (cross-compile on Linux)
-OS X: XCode 4 or Homebrew or GNU Autotools
+OS X: XCode 4 or Homebrew or CMake
 
 The Autotools build system is usually the most up-to-date, and is what is used
```
Comment (impaktor) [inline COMPILING.txt]: Talks about autotools here.

File: COMPILING.txt [hunk 2]
```diff
@@ -49,7 +42,7 @@ If you're having trouble compiling, please ask in #pioneer on
 irc.freenode.net.
```
Comment (impaktor) [inline COMPILING.txt]: While you're at it, you could add "(most users are in CET zone)", even if it's not true for you. Not uncommon to see in my backlog people who enter #pioneer at 2 AM asking for help.

File: COMPILING.txt [hunk 3]
```diff
@@ -58,47 +51,28 @@ irc.freenode.net.
...
-   pioneer-thirdparty repository, pass
-   --with-thirdparty=/path/to/pioneer-thirdparty to configure.
-
-3. Run make to build everything
-
-1.1.b Linux - CMake
--------------------
-
-1. Run './bootstrap cmake' to generate CMake build files in the 'build' folder.
+2. Run './bootstrap cmake' to generate CMake build files in the 'build' folder.
```
Comment (impaktor) [inline COMPILING.txt]: just checking: does bootstrap still need the `cmake` argument?

File: COMPILING.txt [hunk 4]
```diff
@@ -267,16 +158,21 @@ The pioneer-thirdparty repository contains:
...
 
-2.1 Linux - Autotools
+2.1 Linux - Autotools and CMake
 ---------------------
 
-1. After cloning the repo, run autoconf to generate 'configure'.
+1. Clone the thirdparty repo to './pioneer-thirdparty' and 'cd' into it.
 
-2. Run ./configure to configure the build
+1. Run 'autoconf' to generate configuration files, then run './configure' to
```
Comment (impaktor) [inline COMPILING.txt]: Numbering is off here, you have two "ones"

File: COMPILING.txt (hunk 3 above)
Comment (Web-eWorks) [reply COMPILING.txt]: It's supported but not required - I'll remove the mention of it here.

**Comment by orbea** [comment]
It'd still be nice to see a fix for the model compiler paths so that it doesn't escape the DESTDIR. This will occur by setting `-DPIONEER_DATA_DIR=` to any value other than `/usr/share/pioneer`.

**Comment by Web-eWorks** [comment]
@orbea While I'd like to get that done, I'm not sure this PR is related to that issue at all. You'd be better served creating or commenting on a more appropriate issue that's related to the modelcompiler.

**Comment by orbea** [comment]
@Web-eWorks Its related to the cmake implementation because its a regression that was not present in the autotools implementation.

**Pull request merged**